[2/3] Add robocup grammar

ar13pit opened the pull request:
New cfg branch as the old one became extremely dirty and there was no point in keep the history there.

This branch will be made as pythonic as possible.

Resolves tue-robotics/tue_robocup#669

ar13pit committed:
cleaned cfg branch

ar13pit committed:
cleaned up

ar13pit committed:
keep fsFile and svg in bin

ar13pit committed:
added bin and build

ar13pit committed:
unused variable in the sentence rule

ar13pit committed:
fixed script to generate random sentences

ar13pit committed:
auto generated from python script

ar13pit committed:
converted script into object oriented code

ar13pit committed:
added missing quotes

ar13pit committed:
adding FstCompiler class

ar13pit committed:
merged FsSymbolGenerator class with FstCompiler

ar13pit commented:
CWD is tutorial

ar13pit committed:
ignore bin

ar13pit committed:
added check for fpath and new warning method

ar13pit committed:
added outpath

ar13pit committed:
renamed fstcompiler class file

ar13pit committed:
added original makeSymbols file

ar13pit committed:
fixed echo statements

ar13pit committed:
moved tutorial to tutorial-1

ar13pit committed:
tutorial-2

ar13pit committed:
added word extraction from grammar

ar13pit committed:
added creation of file

ar13pit committed:
forked from voxforge

ar13pit committed:
made path to current grammar rel to script

ar13pit committed:
use only lm parts

ar13pit committed:
commented copying of txt files to data

ar13pit committed:
added grammar.py to make corpus

ar13pit committed:
added corpus creator

ar13pit committed:
added echo statements to checks

ar13pit committed:
added conf from voxforge

ar13pit committed:
do not write vocab file

ar13pit committed:
vocab from training data

ar13pit committed:
moved creation of dict up in the chain

ar13pit committed:
added model creation

ar13pit committed:
dragonfly lib no longer needed

ar13pit committed:
Merge remote-tracking branch 'origin/master' into ar13pit/cfg2

samaleksandrov committed:
Created grammar file for extracting the grammar tree.

ar13pit committed:
zipped fst files to work on later

ar13pit commented on egs/cfg2/wip/kaldi_grammar.py:
```diff
@@ -0,0 +1,121 @@
+
+        self.target = target
+
+    def get_rule_element(self, depth=0):
```
This will need the target argument so make the function:
`def get_rule_element(self, target, depth=0)`

samaleksandrov replied:
done

ar13pit commented on egs/cfg2/wip/kaldi_grammar.py:
```diff
@@ -0,0 +1,121 @@
+        """
+        RulesList = {}
+        # print(depth)
+        if self.target not in self.parser.rules:
```
Remove self from targets throughout the function

samaleksandrov replied:
done

ar13pit commented on egs/cfg2/wip/kaldi_grammar.py:
```diff
@@ -0,0 +1,121 @@
+
+                # If variable: go one level deeper
+                if conj.is_variable:
+                    result = get_rule_element(conj.name, self.parser, depth + 1)
```
1. Add self to the function name -> `self.get_rule_element`
2. Remove self.parser from argument

samaleksandrov replied:
done

ar13pit commented on egs/cfg2/wip/kaldi_grammar.py:
```diff
@@ -0,0 +1,121 @@
+    target = sys.argv[2]
+
+    kaldi_gr = KaldiGrammar(grammar_file, target)
+    tree = kaldi_gr.get_rule_element()
```
Add target as an argument to the function

samaleksandrov replied:
done

ar13pit commented:
@SamAlexandrov Make sure you start having more commits rather than just one at the end of the day. It becomes easy to track changes. Make commits after every significant change

ar13pit committed:
removed dragonfly files

samaleksandrov committed:
resolved the comments and tested the tree generation. The tree generation is successful, len(tree) = 3

ar13pit committed:
custom test

ar13pit committed:
reduced number of files by grouping in tar

samaleksandrov committed:
test script for the expand tree function

samaleksandrov committed:
commented the rules for more than one sentence to make the testing of the tree expansion easier. Will be uncommented in future ommit

samaleksandrov committed:
WIP: added expand_tree function and TODO section on what is needed for correct tree traversal

samaleksandrov committed:
added function check_word, which checks if the first word of the sentence matches the recognised word and filters the list of sentences. Updated some incorrect words in the function expand_tree

MatthijsBurgh commented on egs/cfg2/wip/expand_tree_test.py:
```diff
@@ -1,4 +1,6 @@
\ No newline at end of file
+# words = k.expand_tree()
+recognised_word = 'say'
+filter = k.check_word(recognised_word)
```
Don't use filter as variable name as it is an internal function of python.

samaleksandrov replied:
This script is only for testing. After the last commit the file will be deleted (but good remark, anyway) :)

samaleksandrov committed:
WIP: added function autocomplete, which traverses the tree and narrows down the possible next words. Currently implemented with a raw_input, for easy testing, future updates - replaced raw_input with output from the speech, expansion of the whole tree, not only the first element and adding option to skip the recognition of a single word and to jump to the next

samaleksandrov committed:
WIP: added traversal through the whole tree

Still work in progress, added classes, creating the nodes and edges of the grammar tree. Modified function expand_tree, to allow expanding the tree on deeper levels.

samaleksandrov committed:
updated current grammar rules file for testing

samaleksandrov committed:
WIP: functional tree traversal

expands the whole tree and traverses successfully, however - too slow, optmization in progress

ar13pit committed:
Merge remote-tracking branch 'origin/master' into ar13pit/cfg2

alberth committed:
Merge nodes for equal rule suffixes

Commit:
Update kaldi_grammar.py

ar13pit committed:
changed main function to print graphiz node

ar13pit committed:
replaced expand_tree_test with print_tree_pdf

Got rid of the python file. Wrote a bash script to capture the dot file and convert it into pdf

ar13pit committed:
cleaned up commented stuff

ar13pit committed:
removed print_tree

ar13pit committed:
removed legacy get_rule_element method

ar13pit committed:
removed side comments, changed expand tree function to method
